lib/shopwareStoreCommander.js

```diff
@@ -0,0 +1,364 @@
+            return this.loadExtraPluginFields(accountData.plugins[pluginName], extraFields);
+        }
+
+        return accountData.plugins[pluginName];
+    }
+
+    /**
+     * @param {Object} plugin
+     * @return {Promise}
+     */
+    async savePlugin(plugin) {
+        this.logger.emit('info', `Saving changes in plugin ${plugin.name}...`);
+
+        const res = await this.client.put(`plugins/${plugin.id}`, plugin);
+        // Save the updated data locally
+        this.updatePluginData(this.accountData.plugins[plugin.name], res.data);
+
+        return this.accountData.plugins[plugin.name];
+    }
+
+    /**
+     * @param {Object} plugin
+     * @param {String} filePath
+     * @return {Promise}
+     */
+    async uploadPluginBinary(plugin, filePath) {
+        const binaryName = filePath.split(/(\\|\/)/g).pop();
+        this.logger.emit('info', `Uploading binary ${binaryName} for plugin ${plugin.name}...`);
+
+
+        const { data, headers } = await getPostData(filePath);
+        const res = await this.client.post(`plugins/${plugin.id}/binaries`, data, { headers });
+
+        // Add the binary info to the plugin
+        plugin.binaries = res.data;
+        plugin.latestBinary = plugin.binaries[plugin.binaries.length - 1];
+
+        return plugin;
+    }
+
+    /**
+     * @param {Object} plugin
+     * @param {Object} binary
+     * @param {String} filePath
+     * @return {Promise}
+     */
+    async updatePluginBinary(plugin, binary, filePath) {
+        const binaryName = filePath.split(/(\\|\/)/g).pop();
+        this.logger.emit('info', `Uploading updated binary ${binaryName} for plugin ${plugin.name}...`);
+
+        const { data, headers } = await getPostData(filePath);
+        const res = await this.client.post(`plugins/${plugin.id}/binaries/${binary.id}/file`, data, { headers });
+
+        // Add the binary info to the plugin
+        plugin.binaries = res.data;
+        plugin.latestBinary = plugin.binaries[plugin.binaries.length - 1];
+
+        return plugin;
+    }
+
+    /**
+     * @param {Object} plugin
+     * @param {Object} binary
+     * @return {Promise}
+     */
+    async savePluginBinary(plugin, binary) {
+        this.logger.emit('info', `Saving binary version ${binary.version} of plugin ${plugin.name}...`);
+
+        const res = await this.client.put(`plugins/${plugin.id}/binaries/${binary.id}`, binary);
+        // Save the updated data locally
+        binary.changelogs = res.data.changelogs;
+        binary.compatibleSoftwareVersions = res.data.compatibleSoftwareVersions;
+        binary.status = res.data.status;
+
+        return plugin;
+    }
+
+    /**
+     * @param {Object} plugin
+     * @return {Promise}
+     */
+    async requestBinaryReview(plugin) {
+        // Create a function for polling the review status every 3 seconds until review is finished
+        let pollCount = 0;
+        const pollReviewStatus = async (review) => {
```

Comment:
I think a `do...while` loop would be more idiomatic here. Also maybe extract this into a separate method.